haystack_experimental/super_components/indexers/document_indexer.py

```diff
@@ -0,0 +1,200 @@
+# SPDX-FileCopyrightText: 2022-present deepset GmbH <info@deepset.ai>
+#
+# SPDX-License-Identifier: Apache-2.0
+
+from typing import Any, Dict, List, Literal, Optional
+
+from haystack import Pipeline, component, default_from_dict, default_to_dict
+from haystack.components.embedders import SentenceTransformersDocumentEmbedder
+from haystack.components.writers import DocumentWriter
+from haystack.document_stores.types import DocumentStore, DuplicatePolicy
+from haystack.utils import (
+    ComponentDevice,
+    Secret,
+    deserialize_document_store_in_init_params_inplace,
+    deserialize_secrets_inplace,
+)
+from haystack.utils.hf import deserialize_hf_model_kwargs, serialize_hf_model_kwargs
+
+from haystack_experimental.core.super_component import SuperComponent
+
+
+@component
+class DocumentIndexer(SuperComponent):
```

Comment:
Why not `SentenceTransformerDocumentIndexer`? It'll be a bit confusing for people who want to use embedding models from other providers not available via sentence-transformers architecture....